python/tests/test_topology.py

```diff
@@ -4984,3 +4985,236 @@ def test_ts_single_tree_delete_middle(self):
+        nodes (could be sequential mutations)
+        """
+        tables = ts.dump_tables()
+        site = tables.sites.add_row(position, ancestral_state)
+        for state, node in zip(derived_states, nodes):
+            tables.mutations.add_row(site, node, state)
+        tables.sort()
+        tables.build_index()
+        tables.compute_mutation_parents()
+        return tables.tree_sequence()
+
+    def verify_sites(self, source_tree, trimmed_tree, position_offset):
+        source_sites = list(source_tree.sites())
+        trimmed_sites = list(trimmed_tree.sites())
+        self.assertEqual(len(source_sites), len(trimmed_sites))
+        for source_site, trimmed_site in zip(source_sites, trimmed_sites):
+            self.assertAlmostEqual(
+                source_site.position, position_offset + trimmed_site.position)
+            self.assertEqual(
+                source_site.ancestral_state, trimmed_site.ancestral_state)
+            self.assertEqual(source_site.metadata, trimmed_site.metadata)
+            self.assertEqual(
+                len(source_site.mutations), len(trimmed_site.mutations))
+            for source_mut, trimmed_mut in zip(
+                    source_site.mutations, trimmed_site.mutations):
+                self.assertEqual(source_mut.node, trimmed_mut.node)
+                self.assertEqual(
+                    source_mut.derived_state, trimmed_mut.derived_state)
+                self.assertEqual(
+                    source_mut.metadata, trimmed_mut.metadata)
+                # mutation.parent id may have changed after deleting redundant mutations
+                if source_mut.parent == trimmed_mut.parent == tskit.NULL:
+                    pass
+                else:
+                    self.assertEqual(
+                        source_tree.tree_sequence.mutation(source_mut.parent).node,
+                        trimmed_tree.tree_sequence.mutation(trimmed_mut.parent).node)
+
+    def verify_ltrim(self, source_ts, trimmed_ts):
+        self.assertEqual(source_ts.num_trees, trimmed_ts.num_trees + 1)
+        left = source_ts.first().interval[1]
+        for j in range(trimmed_ts.num_trees):
+            source_tree = source_ts.at_index(j + 1)
+            trimmed_tree = trimmed_ts.at_index(j)
+            self.assertEqual(source_tree.parent_dict, trimmed_tree.parent_dict)
+            self.assertAlmostEqual(source_tree.span, trimmed_tree.span)
+            self.assertAlmostEqual(
+                source_tree.interval[0], left + trimmed_tree.interval[0])
+            self.verify_sites(source_tree, trimmed_tree, left)
+
+    def verify_rtrim(self, source_ts, trimmed_ts):
+        self.assertEqual(source_ts.num_trees, trimmed_ts.num_trees + 1)
+        for j in range(trimmed_ts.num_trees):
+            source_tree = source_ts.at_index(j)
+            trimmed_tree = trimmed_ts.at_index(j)
+            self.assertEqual(source_tree.parent_dict, trimmed_tree.parent_dict)
```

Comment:
Missing check for the intervals here. Should be identical, right? (Could probably use standard equality in the case?)

Reply:
Also not checking the overall sequence length (should probably be checked above in ltrim as well)

Reply:
Yep to testing for absolute equality in `verify_rtrim`.

Sequence length will have changed after trimming, so can't be (easily) verified in `verify_ltrim` or `verify_rtrim`. Easiest to understand if this is simply checked in each function (I'll add it)

Reply:
Yes, for testing absolute equality in `rtrim`.

Will add something in to check seq length too